python/src/rest_api/rest_api_impl.py

```diff
@@ -122,6 +122,7 @@ def _write_openapi_spec_to_file(output_directory: str) -> str:
                 routes=app.routes,
             ),
             output_file,
+            indent=4,
         )
 
     return f"Successfully wrote the OpenAPI spec to {full_file_path}."
```

Comment:
So, looking at the remainder of this method and where it is called within the command method, there is no handling for if the JSON dump fails. That would be useful to have in place for better error handling, and would allow for the easier avenue of a failure test path.

Reply:
The json.dump does not use any input from the command, it uses the get_openapi function from fastapi to pull the API data.  I don't think there is any way to influence what json.dump() receives through the tests.

Reply:
Far as impacting the input to the JSON dump, right it's handled by the `get_openapi` far as generating the input to be converted to JSON. But, can do handling for if the action of the dump itself fails. So, something like a `try/except` 

Or, the input to the `get_openapi` is all provided from the generated `app` instance. Can do a check about if that properly exists before trying to do a dump of its content.

Reply:
We could do that, but there would be no way to trigger that exception from the tests.  And the only reason app wouldn't exist in this context would be if you edited the implementation files to remove it, unless I am misunderstanding something.  This command is simple enough to be fairly fool proof as far as I can see.  If something fails here, that means the entire plugin is broken.

Reply:
Looked back at the `aac_rest_app` file that makes the app to be referenced here in implementation and can see what you are saying about the layers of access to make it not exist from in a coded way. And it is a simple command, I could be overcomplicating by wanting to account for situations that might not occur at this level, but like you said be at the higher plugin itself level.

Reply:
I still really think at a minimum of putting a `try/except` around the dump would be good to have just in case and would help bubble out when an import/install of fastapi failed out.

Something like this, and that except would bubble out the message and failure status to put with the plugin execution. 
https://stackoverflow.com/questions/23726289/python-check-if-json-dumps-is-possible

Reply:
I included a try/except to the write function.